### PR TITLE
Add filtering script with options: basic, stars, comments

### PR DESCRIPTION
**Pull request by loubnabnl**
This PR adds filtering based on stars and comment to code ratio (in addition to basic codex filters) + loggings. One can also choose the filters to execute in the run command.

### CODE REVIEW

File: preprocessing/arguments.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
+from dataclasses import dataclass, field
+from typing import Optional
+
+
+@dataclass
+class EvalArguments:
```
Comment (lvwerra) [inline preprocessing/arguments.py]: seems like an odd name, no? maybe `FilteringArguments`?

File: preprocessing/README.md [hunk 2]
```diff
@@ -1,11 +1,16 @@
 # Filtering The Stack (Python/Java/Javascript subsets)
-We use the following filters:
+Three filters for the preprocessing of The Stack are available:
 
-* max line length > 1000
-* mean line length > 100
-* fraction of alphanumeric characters < 0.25
+* **basic**: uses line length filtering and percentage of alphanumeric characters (similarily to [Codex](https://arxiv.org/abs/2107.03374)), default thresholds are `max_line_length=1000`, `mean_line_length=100`, `alpha_num_threshold=0.25`.
+* **stars**: filter based on number of stars of the file (i.e. of the parent repository), default threshold is `threshold_stars=5`.
+* **comments**:  filter based on minimum and maximum comments to code ratio, default thresholds are `min_comments_ratio=0.01`, `max_comments_ratio=0.8`.
```
Comment (lvwerra) [inline preprocessing/README.md]: maybe we can add a little more information on how comments are detected and measured.

File: preprocessing/README.md [hunk 3]
```diff
@@ -1,11 +1,16 @@
...
+Below is an example for running all filters on the java subset of [The-Stack-Smol](https://huggingface.co/datasets/bigcode/the-stack-smol). You can specify the filters to use separated by commas in `--filters` argument:
 ```bash
-python fliltering.py --dataset_name <dataset_name> --subset <subset> --split <split> --hub_username <hub_username> --remote_repo <remote_repo>
-```
\ No newline at end of file
+python filtering.py --dataset_name bigcode/the-stack-smol --subset data/java --filters basic,stars,comments --hub_username loubnabnl --remote_repo test_filter_pipeline_java
+```
+In this case, the dataset is saved in parquet shards in a clone of `remote_repo`. If you want to push the data directly to the hub add `push_to_hub` flag. 
+
+For more details about other arguments check `arguments.py` file.
```
Comment (lvwerra) [inline preprocessing/README.md]: isn't there a `--help` command?

**Review by lvwerra — APPROVED** [review]
Looks good to me. Added a few minor comments.

File: preprocessing/arguments.py (hunk 1 above)
Comment (loubnabnl) [reply preprocessing/arguments.py]: silly copy-paste error :stuck_out_tongue:

File: preprocessing/README.md (hunk 3 above)
Comment (loubnabnl) [reply preprocessing/README.md]: yes, I will remove that line.

**Pull request merged**